### ISSUE TITLE
Non-static default field names result in an invalid QName error

### ISSUE THREAD

**Issue by wemrysi** [quasar-analytics/quasar#1760]
The current workaround of prefixing `_` to the default numeric field names doesn't work when the name isn't a literal number, like in the following query `select meta(*).award, count(*) from oscars group by meta(*).award`.

Ideally #1642 will be fixed first, rendering this moot.

**Comment by wemrysi** [comment]
The example query given actually works now. I think this is still an issue in theory, but possibly more rare that it was before. If I run into another example during testing I'll add it here.

**Comment by wemrysi** [comment]
Here's one from `groupByFlatten.test` that triggers the issue

```
select substring(parents[*].sha, 0, 1), count(*) from slamengine_commits group by substring(parents[*].sha, 0, 1)
```

**Comment by jsantos17** [comment]
Closed via #2275
